MapboxSearch/sdk/src/main/java/com/mapbox/search/result/SearchResult.kt

```diff
@@ -206,6 +214,7 @@ public class SearchResult internal constructor(
         result = 31 * result + (routablePoints?.hashCode() ?: 0)
         result = 31 * result + (boundingBox?.hashCode() ?: 0)
         result = 31 * result + (categories?.hashCode() ?: 0)
+        result = 31 * result + (categoryIds?.hashCode() ?: 0)
```

Comment:
nit: use extension `hashCode` function that takes into account nullable type 
```suggestion
        result = 31 * result + categoryIds.hashCode()
```